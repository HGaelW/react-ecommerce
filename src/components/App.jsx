import React, { Component } from 'react';
import propTypes from 'prop-types';
import Routes from './Routes';
import { connect } from 'react-redux';
import { fetchCategories, fetchDepartments } from '../actions/itemActions';
import { fetchShippingRegions } from '../actions/shippingActions';
import { fetchCurrentUser } from '../actions/currentUserActions';
import './App.scss';
import { fetchAllTax, fetchOrders } from '../actions/orderActions';

export class App extends Component {
  componentDidMount() {
    this.props._fetchCategories();
    this.props._fetchDepartments();
    this.props._fetchCurrentUser();
    this.props._fetchAllTax();
    this.props._fetchShippingRegions();
    this.props._fetchOrders();
  }
  render() {
    return <Routes />;
  }
}

App.propTypes = {
  getCategories: propTypes.func,
  getDepartments: propTypes.func,
};

export const mapDispatchToProps = dispatch => ({
  _fetchCategories: () => dispatch(fetchCategories()),
  _fetchDepartments: () => dispatch(fetchDepartments()),
  _fetchCurrentUser: () => dispatch(fetchCurrentUser()),
  _fetchAllTax: () => dispatch(fetchAllTax()),
  _fetchShippingRegions: () => dispatch(fetchShippingRegions()),
  _fetchOrders: () => dispatch(fetchOrders()),
});

export default connect(
  null,
  mapDispatchToProps,
)(App);
