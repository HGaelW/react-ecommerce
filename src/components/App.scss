body {
  div {
    color: blue;
    text-align: center;
  }
}
