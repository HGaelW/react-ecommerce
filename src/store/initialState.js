module.exports = {
  currentUser: {
    loginModel: false,
    isAuth: localStorage.getItem('token') || false,
    user: {},
  },
  item: {
    loadingAllItems: false,
    loadingItems: false,
    allItems: [],
    singleItem: {},
  },
};
