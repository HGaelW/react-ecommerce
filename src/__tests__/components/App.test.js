import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import App from '../../components/App';
import initialState from '../../store/initialState';

let wrapper;
const mockStore = configureMockStore([]);
const store = mockStore(initialState);

describe('App.jsx', () => {
  test('should render App.jx', () => {
    wrapper = shallow(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
