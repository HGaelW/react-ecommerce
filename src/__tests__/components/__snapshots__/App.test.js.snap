// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.jsx should render App.jx 1`] = `ShallowWrapper {}`;
